spec/puppet-server-lib/puppet/jvm/http_client_spec.rb

```diff
@@ -21,4 +21,19 @@
       expect(request_options.get_connect_timeout_milliseconds).to equal(connect_timeout)
     end
   end
+
+  context "requests" do
+    it "throw Ruby SocketErrors instead of Java exceptions" do
+      client = Puppet::Server::HttpClient.new("localhost", 0, {})
+      [lambda { client.get('/', nil) },
+       lambda { client.post('/', nil, nil) }
+      ].each do |request_block|
+        expect(&request_block).to raise_error { |error|
+          expect(error).to be_a(SocketError)
+          expect(error).to be_a(Puppet::Server::HttpClientError)
+          expect(error.message).to eq('com.puppetlabs.http.client.HttpClientException: Error executing http request')
+        }
```

Comment:
Yeah, I was assuming we might need to do something like `with-log-output-atom` - https://github.com/puppetlabs/trapperkeeper/blob/trapperkeeper-1.1.0/test/puppetlabs/trapperkeeper/testutils/logging.clj#L88-L106 - in trapperkeeper's testutils does to stop current loggers, run the test functionality you don't want to have processed by the console appender, and restart the original loggers when done.  Maybe as a Ruby helper method that these spec tests could call as desired?  Definitely not worth holding this PR up, though.

Someday we'll tell future generations about the day @nwolfe intentionally violated his own principle.  :smile: